games-emulation/higan/higan-116_pre.ebuild

```diff
@@ -0,0 +1,138 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop toolchain-funcs xdg
+
+DESCRIPTION="A multi-system emulator focused on accuracy, preservation, and configurability"
+HOMEPAGE="https://github.com/higan-emu/higan"
+SRC_URI="https://github.com/higan-emu/${PN}/archive/nightly.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="
+	alsa ao +byuu +gtk higan-ui icarus openal +opengl oss +pulseaudio +sdl udev
+	xv"
+REQUIRED_USE="|| ( byuu higan-ui icarus )"
```

Comment:
If it is correct that `alsa`, `ao`, `openal`, `opengl`, `pulseaudio`, `sdl`, `udev` and `xv` flags are only effective with either `byuu` or `higan-ui` then you'll need to add a bunch of additional constraints not to make them ambiguous. Unless it is a mistake -  at least previous ebuilds did not make that distinction.

Reply:
For some historical context: previous versions of higan did not include `byuu` which is a new alternative user interface for casual users that don't need the advanced controls of the full `higan-ui` user interface; `icarus` is [a separate tool bundled with higan that allows you to bulk-import ROM files into higan’s game library](https://higan.readthedocs.io/en/latest/interface/icarus/).

Because `icarus` is just a game library management tool, the driver USE flags (`alsa`, `ao`, `openal`, etc..) don't apply to it. I believe `icarus` is intended to be run in conjunction with the `higan-ui` user interface, but I created a USE flag option for it because the tool could in theory be run independently since it's just used to construct and maintain the layout of your games directory on your filesystem.

If you think it'll be simpler for users, I can remove the `icarus` flag and build it only when `higan-ui` is selected -- this would match how upstream bundles `icarus` with `higan-ui` while `byuu` is alone.

Reply:
Given that `icarus` flag also does not drive any additional dependencies, it makes sense to build it unconditionally and (together with `byuu`) drop the REQUIRED_USE complexity.